staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1046,77 +1046,12 @@ type VersionedFieldSet struct {
 	// set because it cannot be automatically converted.
 	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,1,opt,name=apiVersion"`
 	// Fields identifies a set of fields.
-	Fields FieldSet `json:"fields,omitempty" protobuf:"bytes,2,opt,name=fields,casttype=FieldSet"`
+	Fields Fields `json:"fields,omitempty" protobuf:"bytes,2,opt,name=fields,casttype=Fields"`
```

Comment:
Do we want to make it `omitempty` or do we want to keep the ability to differentiate between completely removed and set-to-null/empty? We could also differentiate based on the lack of "VersionedFieldSet" vs lack of Fields.